semgrep/semgrep/core_runner.py

```diff
@@ -272,6 +261,80 @@ def get_files_for_language(
             ) from ex
         return list(targets)
 
+    def _get_targets(
+        self, rules: List[Rule], target_manager: TargetManager, all_targets: Set[Path]
+    ) -> List[Dict[str, Any]]:
+        target_info: Dict[Tuple[Path, Language], List[RuleId]] = {}
+
+        for rule in rules:
+            for language in rule.languages:
+                targets = self.get_files_for_language(language, rule, target_manager)
+
+                for target in targets:
+                    all_targets.add(target)
+                    t = (target, language)
+                    if t in target_info:
+                        target_info[t].append(RuleId(rule.orig_id))
+                    else:
+                        target_info[t] = [RuleId(rule.orig_id)]
```

Comment:
Also potentially cleaner if in the inner for loop you construct a list and after that add it to target_info instead of constantly updating in place.

Reply:
Do we also need to worry about name collisions with orig_id

Reply:
Not certain I see how this works. My outside loop is iterating rule-wise but I'm building a dictionary target-wise